src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1781,43 +1779,40 @@ class Simplex_tree {
     Vertex_handle maxvert = std::numeric_limits<Vertex_handle>::min();
     Filtration_value minval = std::numeric_limits<Filtration_value>::infinity();
     Filtration_value maxval = -std::numeric_limits<Filtration_value>::infinity();
-    for (auto sh = root_.members().begin(); sh != root_.members().end(); ++sh){
+    for (auto sh = root_.members().begin(); sh != root_.members().end(); ++sh) {
       Filtration_value f = this->filtration(sh);
       minval = std::min(minval, f);
       maxval = std::max(maxval, f);
       maxvert = std::max(sh->first, maxvert);
     }
     
     GUDHI_CHECK(maxvert < std::numeric_limits<Vertex_handle>::max(), std::invalid_argument("Simplex_tree contains a vertex with the largest Vertex_handle"));
-    maxvert += 1;
+    maxvert++;
 
     Simplex_tree st_copy = *this;
 
     // Add point for coning the simplicial complex
     this->insert_simplex_raw({maxvert}, -3);
 
-    // For each simplex
-    std::vector<Vertex_handle> vr;
-    for (auto sh_copy : st_copy.complex_simplex_range()){
+    Filtration_value scale = maxval-minval;
+    if (std::fabs(maxval - minval) > std::numeric_limits<Filtration_value>::epsilon())
```

Comment:
Comparing a value to epsilon does not make sense to me. epsilon is for relative stuff, or bit manipulations.

Reply:
Done on [8550ad3](https://github.com/GUDHI/gudhi-devel/pull/888/commits/8550ad3b01f047a30044cbcce8c142a421da96f4)